### PR TITLE
Add half porstion

### PR DESCRIPTION
**Pull request by TopOleh**

### CODE REVIEW

File: src/app/core/services/firebase.service.ts [hunk 1]
```diff
@@ -102,7 +103,7 @@ export class FirebaseService {
         })
       );
   }
-  
+
   addOrder(orderDetails: OrderDetails): Promise<firebase.firestore.DocumentReference> {
```
Comment (TsirohOleh) [inline src/app/core/services/firebase.service.ts]: public

File: src/app/modules/admin-page/admin-page.component.ts [hunk 2]
```diff
@@ -11,38 +11,67 @@ import { FormBuilder, FormGroup, Validators, FormControl } from '@angular/forms'
...
+  @Input() complex;
+  @Input() name;
+  @Input() half;
+  @Input() type;
+  @Input() todaymenu;
+  @Input() price;
+  @Input() weight;
+  @Input() img;
+  @Input() info;
+  @Input() id;
```
Comment (TsirohOleh) [inline src/app/modules/admin-page/admin-page.component.ts]: @Input()
dish: Dish;

File: src/app/shared/menu-item/menu-item.component.html [hunk 3]
```diff
@@ -25,94 +37,16 @@ <h5 class="card-title text-center my-2">{{dish.name}}</h5>
...
-        <input type="file" (change)="addImage($event)" class="custom-file-input" id="image" required>
-        <label class="custom-file-label" for="image">Choose image...</label>
-        <div class="invalid-feedback">Choose image for your dish</div>
-      </div>
-    </form>
-  </div>
-  <div class="modal-footer">
-    <button class="btn btn-primary" (click)="saveEditDish()" type="submit">Submit form</button>
-  </div>
+  <app-admin-page
```
Comment (TsirohOleh) [inline src/app/shared/menu-item/menu-item.component.html]: [dish]="dish"

File: src/app/modules/admin-page/admin-page.component.ts [hunk 4]
```diff
@@ -11,38 +11,67 @@ import { FormBuilder, FormGroup, Validators, FormControl } from '@angular/forms'
...
+    this.dish.id = this.id               || this.dish.id;
+    this.dish.complex = this.complex     || this.dish.complex;
+    this.dish.half = this.half           || this.dish.half;
+    this.dish.name = this.name           || this.dish.name;
+    this.dish.type = this.type           || this.dish.type;
+    this.dish.todaymenu = this.todaymenu || this.dish.todaymenu;
+    this.dish.price = this.price         || this.dish.price;
+    this.dish.weight = this.weight       || this.dish.weight;
+    this.dish.img = this.img             || this.dish.img;
+    this.dish.info = this.info           || this.dish.info;
```
Comment (TsirohOleh) [inline src/app/modules/admin-page/admin-page.component.ts]: Why do you need it?

Comment (TopOleh) [reply src/app/modules/admin-page/admin-page.component.ts]: If we have dish from input we use that info in other case existing data

**Pull request merged**